modules/unsupported/process-feature/src/main/java/org/geotools/process/vector/ClassifyByRangeProcess.java

```diff
@@ -0,0 +1,144 @@
+            if(classifyDescriptor == null) {
+                throw new ProcessException("classifyOnAttribute is not a valid schema attribute: " + classifyOnAttribute);
+            }
+            Class<?> rangeType = classifyDescriptor.getType().getBinding();
+            if(rangeType == null) {
+                rangeType = Number.class;
+            }
+            List<Expression> params = new ArrayList<Expression>();
+            params.add(filterFactory.property(classifyOnAttribute));
+            if((thresholds == null || thresholds.trim().isEmpty()) && (classifier == null || classifier.trim().isEmpty())) {
+                throw new ProcessException("at least one of thresholds and classifier is mandatory");
+            }
+            List<Object> ranges = new ArrayList<Object>();
+            if(classifier!= null && !classifier.trim().isEmpty()) {
+                if (classes == null) {
+                    classes = 5;
+                }
+                Function classifyFun = filterFactory.function(classifier, filterFactory.property(classifyOnAttribute), 
+                        filterFactory.literal(classes));
+                RangedClassifier rc = (RangedClassifier)classifyFun.evaluate(features);
+                for (int i = 0; i < rc.getSize(); i++) {
+                    ranges.add(rc.getMin(i));
+                }
+                ranges.add(rc.getMax(rc.getSize() - 1));
+            } else {
+                for(String threshold : thresholds.split(",")) {
+                    ranges.add(threshold);
+                }
+            }
+                
+            if(outputValues == null || outputValues.trim().isEmpty()) {
+                outputValues = "";
+                for(int count = 1; count <= ranges.size();count++) {
+                    outputValues += count + ",";
+                }
+                outputValues += (ranges.size() + 1 + "");
+            }
+            String[] rangesValues = outputValues.split(",");
+            if (rangesValues.length != (ranges.size() + 1)) {
+                throw new ProcessException("values are not consistent with thresholds (should be +1)");
+            }
+            for(int count = 0; count < ranges.size(); count++) {
+                params.add(filterFactory.literal(Converters.convert(rangesValues[count], classify.binding)));
+                params.add(filterFactory.literal(Converters.convert(ranges.get(count), rangeType)));
+            }
+            params.add(filterFactory.literal(Converters.convert(rangesValues[rangesValues.length - 1], classify.binding)));
+            params.add(filterFactory.literal(include ? CategorizeFunction.PRECEDING : CategorizeFunction.SUCCEEDING));
+            classify.expression = new CategorizeFunction(params, null);
+            
+            
+            transform.add(classify);
+            return new TransformProcess().executeList(features, transform);
+        }
+        throw new ProcessException("features input cannot be null!");
```

Comment:
Small style bit, I'd flip the logic and avoid keeping the core of the process code inside an if block (flatter code is normally easier to read)